rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -295,8 +296,18 @@ impl MultisigCheckpointSyncer {
                         );
                         continue;
                     }
+
                     // Ensure that the signature is actually by the validator
-                    let signer = signed_checkpoint.recover()?;
+                    let domain =
+                        KnownHyperlaneDomain::try_from(signed_checkpoint.value.mailbox_domain)?;
+
+                    let signer = match domain.domain_protocol() {
+                        HyperlaneDomainProtocol::Cosmos => {
+                            pub_to_binary_addr(signed_checkpoint.recover_pubkey()?.to_vec())?
+                        }
+                        _ => signed_checkpoint.recover()?,
+                    };
```

Comment:
should `checkpoint.recover` be a trait on hyperlane-core?

Reply:
In my opinion, yes. The `hyperlane-core` traits have to consider multi-chain environment more than now.